crates/sui-package-management/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context};
+use std::path::PathBuf;
+
+use move_package::lock_file::{self, LockFile};
+use sui_json_rpc_types::{get_new_package_obj_from_response, SuiTransactionBlockResponse};
+use sui_sdk::wallet_context::WalletContext;
+
+pub enum LockCommand {
+    Publish,
+    Upgrade,
+}
+
+/// Update the `Move.lock` file with automated address management info.
+/// Expects a wallet context, the publish or upgrade command, its response.
+/// The `Move.lock` principally file records the published address (i.e., package ID) of
+/// a package under an environment determined by the wallet context config. See the
+/// `ManagedPackage` type in the lock file for a complete spec.
+pub async fn update_lock_file(
+    context: &WalletContext,
+    command: LockCommand,
+    install_dir: Option<PathBuf>,
+    lock_file: Option<PathBuf>,
+    response: &SuiTransactionBlockResponse,
+) -> Result<(), anyhow::Error> {
+    let chain_identifier = context
+        .get_client()
+        .await
+        .context("Network issue: couldn't use client to connect to chain when updating Move.lock")?
+        .read_api()
+        .get_chain_identifier()
+        .await
+        .context("Network issue: couldn't determine chain identifier for updating Move.lock")?;
+
+    let (original_id, version, _) = get_new_package_obj_from_response(response).context("Expected a valid published package response but didn't see one when attempting to update the `Move.lock`.")?;
+    let Some(lock_file) = lock_file else {
+        bail!("Expected a `Move.lock` file to exist after publishing package, but none found. Consider running `sui move build` to generate the `Move.lock` file in the package directory.")
+    };
+    let install_dir = install_dir.unwrap_or(PathBuf::from("."));
+    let env = context
+        .config
+        .get_active_env()
+        .context("Could not resolve environment from active wallet context. Try ensure `sui client active-env` is valid.")?;
```

Comment:
Added new contextual errors. It's difficult to anticipate exactly what would cause, e.g., a wallet context to not work, or a `Move.lock` not to exist at this point, given these are basically necessary for the preceding `publish` command.

I've tried to add helpful commands that "should" work if an error ends up originating here (`sui move build`, `sui client active-env`).

Reply:
> how they can get themselves out of it (in this case it would be find out the addresses themselves and run the manual update command)

I think this should apply especially for the logic detecting, e.g., conflicts where `published-at` is different to (or should be modified or removed) based on `Move.lock` updates. 

We can be more explicit about the manual update command here, but, that's not yet updating the lock file (for one, it needs to be updated to reflect our schema change to support environments). The plan is to do that after this PR--I can come back and update warning messages to reflect that user option then.

Reply:
Yeah fair, we don't need to have a self-service fix/answer for everything (if we really don't expect something to happen, I think the action should be to report the issue to us, so that might apply here).